guide/groovy.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Groovy Script Injection
+---
+
+### Overview
+
+The tool offers the ability to run arbitrary groovy scripts on the sources prior to running the build. This allows GME to be extensible by the user and to carry out non-generic modifications.
+
+
+### Configuration
+
+If the property `-DgroovyScripts=<value>,....` is set, it will load the remote Groovy script file.
+
+The argument should a comma separated list of HTTP / HTTPS URLs.
+
+
+### Groovy Scripts
+
+Each groovy script will be run on the execution root (i.e. where Gradle is invoked).
+
+
+Each script <b>must</b> use the following annotations:
+
+```
+import groovy.transform.BaseScript
+import org.jboss.gm.analyzer.alignment.groovy.GMEBaseScript
+
+@BaseScript GMEBaseScript gmeScript
+
+```
+
+The scripts will be run **after** the initial alignment.
```

Comment:
Perhaps we should say there that the scripts are run as the *last* part of the alignment process?

Reply:
Ok sure